docs/architecture/component_templates.md

```diff
@@ -0,0 +1,134 @@
+# Account component templates
+
+An account component template provides a general description of an account component. It encapsulates all the information needed to initialize and manage the component.
+
+Specifically, a template specifies a component's **metadata** and its **code**.
+
+Once defined, a component template can be instantiated as account components, which can then be grouped to form the account's `Code`.
```

Comment:
```suggestion
Once defined, a component template can be instantiated as account components, which can then be merged to form the account's `Code`.
```

Nit: I would call this process merging, which is consistent with how it is described in some Rust doc comments.